[CI] NodeDeprecationChecksTests testTransportSettings failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/p7r32hchn6faa/tests/:x-pack:plugin:deprecation:test/org.elasticsearch.xpack.deprecation.NodeDeprecationChecksTests/testTransportSettings

**Reproduction line:**
`gradlew ':x-pack:plugin:deprecation:test' --tests "org.elasticsearch.xpack.deprecation.NodeDeprecationChecksTests.testTransportSettings" -Dtests.seed=D81B2F794DC1BAF -Dtests.locale=nn-NO -Dtests.timezone=America/Indiana/Knox -Druntime.java=8`

**Applicable branches:**
7.16

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.deprecation.NodeDeprecationChecksTests&tests.test=testTransportSettings

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a collection containing <{"level":"critical","message":"Setting [network.tcp.connect_timeout] is deprecated","url":"https://ela.st/es-deprecation-7-transport-settings","details":"Remove the [network.tcp.connect_timeout] setting and set [transport.connect_timeout] to [0nanos].","resolve_during_rolling_upgrade":false}>
     but: mismatches were: [was <{"level":"critical","message":"Security is enabled by default for all licenses","url":"https://ela.st/es-deprecation-7-implicitly-disabled-security","details":"Security will no longer be disabled by default for Trial licenses in 8.0. The [xpack.security.enabled] setting will always default to \"true\". See https://ela.st/es-deprecation-7-security-minimal-setup to secure your cluster. To explicitly disable security, set [xpack.security.enabled] to \"false\" (not recommended).","resolve_during_rolling_upgrade":false}>, was <{"level":"critical","message":"Setting [http.tcp_no_delay] is deprecated","url":"https://ela.st/es-deprecation-7-transport-settings","details":"Remove the [http.tcp_no_delay] setting and set [http.tcp.no_delay] to [false].","resolve_during_rolling_upgrade":false}>, was <{"level":"critical","message":"Setting [network.tcp.connect_timeout] is deprecated","url":"https://ela.st/es-deprecation-7-transport-settings","details":"Remove the [network.tcp.connect_timeout] setting and set [transport.connect_timeout] to [0s].","resolve_during_rolling_upgrade":false}>, was <{"level":"critical","message":"Setting [transport.tcp.connect_timeout] is deprecated","url":"https://ela.st/es-deprecation-7-transport-settings","details":"Remove the [transport.tcp.connect_timeout] setting and set [transport.connect_timeout] to [5nanos].","resolve_during_rolling_upgrade":false}>, was <{"level":"critical","message":"Setting [transport.tcp.port] is deprecated","url":"https://ela.st/es-deprecation-7-transport-settings","details":"Remove the [transport.tcp.port] setting and set [transport.port] to [INhPnLiamn].","resolve_during_rolling_upgrade":false}>, was <{"level":"critical","message":"Setting [transport.tcp_no_delay] is deprecated","url":"https://ela.st/es-deprecation-7-transport-settings","details":"Remove the [transport.tcp_no_delay] setting and set [transport.tcp.no_delay] to [false].","resolve_during_rolling_upgrade":false}>, was <{"level":"critical","message":"Setting [transport.tcp.compress] is deprecated","url":"https://ela.st/es-deprecation-7-transport-settings","details":"Remove the [transport.tcp.compress] setting and set [transport.compress] to [TRUE].","resolve_during_rolling_upgrade":false}>]

  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.deprecation.NodeDeprecationChecksTests.testTransportSettings(NodeDeprecationChecksTests.java:2216)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

--- a/x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecksTests.java
+++ b/x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecksTests.java
@@ -2178,11 +2178,11 @@ public class NodeDeprecationChecksTests extends ESTestCase {
         Map<Setting<?>, Object> settingToTestValueMap = new HashMap<>();
         settingToTestValueMap.put(HttpTransportSettings.OLD_SETTING_HTTP_TCP_NO_DELAY, randomBoolean());
         /*
-         * Note: Limiting time values to between 0 and 23 because the downstream code will normalize them so that 100 hours becomes 4.2
+         * Note: Limiting time values to between 1 and 23 because the downstream code will normalize them so that 100 hours becomes 4.2
          * days, so the expected values would not match the actual.
          */
-        settingToTestValueMap.put(NetworkService.TCP_CONNECT_TIMEOUT, randomTimeValue(0, 23));
-        settingToTestValueMap.put(TransportSettings.TCP_CONNECT_TIMEOUT, randomTimeValue(0, 23));
+        settingToTestValueMap.put(NetworkService.TCP_CONNECT_TIMEOUT, randomTimeValue(1, 23));
+        settingToTestValueMap.put(TransportSettings.TCP_CONNECT_TIMEOUT, randomTimeValue(1, 23));
         settingToTestValueMap.put(TransportSettings.OLD_PORT, randomAlphaOfLength(10));
         settingToTestValueMap.put(TransportSettings.OLD_TCP_NO_DELAY, randomBoolean());
         settingToTestValueMap.put(

